### PR TITLE
fix(k8s) Improvements to enterprise K8s installation page

### PR DESCRIPTION
**Pull request by cwurm**
Fixes the command to create the session config secret. `admin_gui_session_conf` and `portal_session_conf` are actually files that have been created in the instructions just before.

### CODE REVIEW

**Comment by cwurm** [comment]
While working with this docs page I created a number of other improvements and fixes. Putting them all in this PR.

They include:
- Expanded parameter table with other necessary or useful parameters (e.g. pullSecrets is actually created but not specified)
- Removing unnecessary whitespace in commands
- Aligning position of `-n kong` parameter to always be the first "dashed" parameter in a command
- Change helm commands from `--namesapce` to `-n` to be consistent with kubectl
- Consistently use the current directory (`./`) when specifying a file

**Comment by lena-larionova** [comment]
@marcoam should take a look at this one too, he was planning on updating this topic. Marco, are these all the parameters you wanted to add to the table?

File: app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 1]
```diff
@@ -160,13 +160,24 @@ In the following steps, replace `<your-password>` with a secure password.
     |Parameter      | Value                         |
     |---------------|-------------------------------|
     |`enterprise.enabled` | `true` |
+    |`enterprise.rbac.enabled` | (Optional) Set to `true` to enable RBAC. Requires seeding the super admin password, see above. |
```
Comment (lena-larionova) [inline app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md:163]:
```suggestion
    |`enterprise.rbac.enabled` | (Optional) Set to `true` to enable RBAC. Requires seeding the super admin password; see [above](#step-5-seed-the-super-admin-password). |
```

File: app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 2]
```diff
@@ -160,13 +160,24 @@ In the following steps, replace `<your-password>` with a secure password.
...
     |`env.database` | `"postgres"` or `"cassandra"` |
+    |`env.pg_host` | (If using Postgres) Set to host of the Postgres server (only if `postgresql.enabled` is `false`). |
+    |`env.pg_port` | (If using Postgres) Set to port of the Postgres server (only if `postgresql.enabled` is `false`). |
+    |`env.pg_user` | (If using Postgres) Set to Postgres user (default `kong`). When `postgresql.enabled` is `true` this has to match `postgresql.postgresqlUsername`. |
```
Comment (lena-larionova) [inline app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md:167]:
```suggestion
    |`env.pg_user` | (If using Postgres) Set to the Postgres user (default `kong`). When `postgresql.enabled` is `true`, this has to match `postgresql.postgresqlUsername`. |
```

File: app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 3]
```diff
@@ -160,13 +160,24 @@ In the following steps, replace `<your-password>` with a secure password.
...
+    |`env.pg_password` | (If using Postgres) Set to Postgres user's password. When `postgresql.enabled` is `true` this has to match `postgresql.postgresqlPassword`. |
```
Comment (lena-larionova) [inline app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md:168]:
```suggestion
    |`env.pg_password` | (If using Postgres) Set to the Postgres user's password. When `postgresql.enabled` is `true`, this has to match `postgresql.postgresqlPassword`. |
```

File: app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 4]
```diff
@@ -160,13 +160,24 @@ In the following steps, replace `<your-password>` with a secure password.
...
+    |`env.pg_database` | (If using Postgres) Set to Postgres database name (default `kong`). When `postgresql.enabled` is `true` this has to match `postgresql.postgresqlDatabase`. |
```
Comment (lena-larionova) [inline app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md:169]:
```suggestion
    |`env.pg_database` | (If using Postgres) Set to the Postgres database name (default `kong`). When `postgresql.enabled` is `true`, this has to match `postgresql.postgresqlDatabase`. |
```

File: app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 5]
```diff
@@ -160,13 +160,24 @@ In the following steps, replace `<your-password>` with a secure password.
...
     |`env.password.valueFrom.secretKeyRef.name` | Name of secret that holds the super admin password. In the example above, this is set to `kong-enterprise-superuser-password`. |
     |`env.password.valueFrom.secretKeyRef.key` | The type of secret key used for authentication. If you followed the default settings in the example above, this is `password`. |
     |`image.repository` | The Docker repository. In this case, `kong-docker-kong-enterprise-edition-docker.bintray.io/kong-enterprise-edition`. |
-    |`image.tag` | The Docker image tag you want to pull down, e.g. `"1.5.0.2-alpine"`. |
+    |`image.tag` | The Docker image tag you want to pull down, e.g. `"1.5.0.6-alpine"`. |
```
Comment (lena-larionova) [inline app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md:173]:
```suggestion
    |`image.tag` | The Docker image tag you want to pull down, e.g. `"{{page.kong_latest.version}}-alpine"`. |
```
Let's make this a variable instead so we don't have to keep updating it.

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 6]
```diff
@@ -171,13 +171,24 @@ In the following steps, replace `<your-password>` with a secure password.
...
+    |`env.pg_password` | (If using Postgres) Set to Postgres user's password. When `postgresql.enabled` is `true` this has to match `postgresql.postgresqlPassword`. |
+    |`env.pg_database` | (If using Postgres) Set to Postgres database name (default `kong`). When `postgresql.enabled` is `true` this has to match `postgresql.postgresqlDatabase`. |
     |`env.password.valueFrom.secretKeyRef.name` | Name of secret that holds the super admin password. In the example above, this is set to `kong-enterprise-superuser-password`. |
     |`env.password.valueFrom.secretKeyRef.key` | The type of secret key used for authentication. If you followed the default settings in the example above, this is `password`. |
     |`image.repository` | The Docker repository. In this case, `kong-docker-kong-enterprise-edition-docker.bintray.io/kong-enterprise-edition`. |
-    |`image.tag` | The Docker image tag you want to pull down, e.g. `"1.5.0.3-alpine"`. |
+    |`image.tag` | The Docker image tag you want to pull down, e.g. `"2.1.4.0-alpine"`. |
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:184]:
```suggestion
    |`image.tag` | The Docker image tag you want to pull down, e.g. `"{{page.kong_latest.version}}-alpine"`. |
```

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 7]
```diff
@@ -171,13 +171,24 @@ In the following steps, replace `<your-password>` with a secure password.
     |Parameter      | Value                         |
     |---------------|-------------------------------|
     |`enterprise.enabled` | `true` |
+    |`enterprise.rbac.enabled` | (Optional) Set to `true` to enable RBAC. Requires seeding the super admin password, see above. |
     |`env.database` | `"postgres"` or `"cassandra"` |
+    |`env.pg_host` | (If using Postgres) Set to host of the Postgres server (only if `postgresql.enabled` is `false`). |
+    |`env.pg_port` | (If using Postgres) Set to port of the Postgres server (only if `postgresql.enabled` is `false`). |
+    |`env.pg_user` | (If using Postgres) Set to Postgres user (default `kong`). When `postgresql.enabled` is `true` this has to match `postgresql.postgresqlUsername`. |
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:178]:
```suggestion
    |`env.pg_user` | (If using Postgres) Set to the Postgres user (default `kong`). When `postgresql.enabled` is `true`, this has to match `postgresql.postgresqlUsername`. |
```

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 8]
```diff
@@ -171,13 +171,24 @@ In the following steps, replace `<your-password>` with a secure password.
...
+    |`env.pg_password` | (If using Postgres) Set to Postgres user's password. When `postgresql.enabled` is `true` this has to match `postgresql.postgresqlPassword`. |
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:179]:
```suggestion
    |`env.pg_password` | (If using Postgres) Set to the Postgres user's password. When `postgresql.enabled` is `true`, this has to match `postgresql.postgresqlPassword`. |
```

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 9]
```diff
@@ -171,13 +171,24 @@ In the following steps, replace `<your-password>` with a secure password.
...
+    |`env.pg_database` | (If using Postgres) Set to Postgres database name (default `kong`). When `postgresql.enabled` is `true` this has to match `postgresql.postgresqlDatabase`. |
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:180]:
```suggestion
    |`env.pg_database` | (If using Postgres) Set to the Postgres database name (default `kong`). When `postgresql.enabled` is `true`, this has to match `postgresql.postgresqlDatabase`. |
```

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 10]
```diff
@@ -171,13 +171,24 @@ In the following steps, replace `<your-password>` with a secure password.
     |Parameter      | Value                         |
     |---------------|-------------------------------|
     |`enterprise.enabled` | `true` |
+    |`enterprise.rbac.enabled` | (Optional) Set to `true` to enable RBAC. Requires seeding the super admin password, see above. |
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:174]:
```suggestion
    |`enterprise.rbac.enabled` | (Optional) Set to `true` to enable RBAC. Requires seeding the super admin password; see [above](#step-5-seed-the-super-admin-password). |
```

**Comment by marcoam** [comment]
Yes, I had actually worked on this already but since this is already here, let's continue on this. What I feel we're missing on this PR Is the following: 
1) In Step 7, we're still referencing a non-KE values.yaml. I would suggest that we change this and refer to this: 
https://github.com/Kong/charts/tree/main/charts/kong/example-values. This is what I had written: 

> 1. Create a `values.yaml` file based on the sample in the [Kong charts repository](https://github.com/Kong/charts/tree/main/charts/kong/example-values). If you want to enable all of Kong Enterprise's functionalities, you should use [this file](https://github.com/Kong/charts/blob/main/charts/kong/example-values/full-k4k8s-with-kong-enterprise.yaml).


2) I also had added the "enterprise" values to the table. For example: enterprise.license_secret, enterprise.vitals.enabled and enterprise.portal.enabled. This is what I had: 

>     |`enterprise.license_secret` | Name of secret that holds the Kong Enterprise license. In the example above, this is set to `kong-enterprise-license`.  |
>     |`enterprise.vitals.enabled` | Set to `true` to enable analytics feature of Kong Enterprise. Find out more about Vitals [here](https://docs.konghq.com/enterprise/latest/vitals/overview/). |
>     |`enterprise.portal.enabled` | Set to `true` to enable Kong Enterprise's Developer Portal. Find out more about Vitals [here](https://docs.konghq.com/enterprise/latest/developer-portal/). |
>     |`enterprise.rbac.enabled` | Set to `true` to enable RBAC (role-based access control) for Kong Enterprise's Manager. |

3) I would also order the parameters on the table based on the order of how they appear on the values.yaml file. 

@cwurm - Let me know your thoughts on these changes and also if you want me to change them on your branch of if you want to do it yourself. I'm good with either.

FYI - @lena-larionova

**Comment by cwurm** [comment]
> 1. In Step 7, we're still referencing a non-KE values.yaml. 

I see your point, though I think we're mainly referencing this one because it is the "master" values file that has all possible configuration properties, most of them commented out. So this is the file that users can look through to find anything they might need. I think it would be nice to have a similar master file with the enterprise configuration turned on by default, but that would be something for the https://github.com/Kong/charts repository.

> I would suggest that we change this and refer to this:
>    [https://github.com/Kong/charts/tree/main/charts/kong/example-values](https://github.com/Kong/charts/tree/main/charts/kong/example-values?rgh-link-date=2020-10-21T07%3A09%3A45Z). This is what I had written:
> 
> > 1. Create a `values.yaml` file based on the sample in the [Kong charts repository](https://github.com/Kong/charts/tree/main/charts/kong/example-values?rgh-link-date=2020-10-21T07%3A09%3A45Z). If you want to enable all of Kong Enterprise's functionalities, you should use [this file](https://github.com/Kong/charts/blob/main/charts/kong/example-values/full-k4k8s-with-kong-enterprise.yaml?rgh-link-date=2020-10-21T07%3A09%3A45Z).

How about this:

1. Create a `values.yaml` file for Helm based on the template in the [Kong charts repository](https://github.com/Kong/charts/blob/main/charts/kong/values.yaml). This file contains all the necessary parameters for your Kong deployment. As an example, [this values file](https://github.com/Kong/charts/blob/main/charts/kong/example-values/full-k4k8s-with-kong-enterprise.yaml) enables most Kong Enterprise features.

> 1. I also had added the "enterprise" values to the table. For example: enterprise.license_secret, enterprise.vitals.enabled and enterprise.portal.enabled. This is what I had:
> 
> > ```
> > |`enterprise.license_secret` | Name of secret that holds the Kong Enterprise license. In the example above, this is set to `kong-enterprise-license`.  |
> > |`enterprise.vitals.enabled` | Set to `true` to enable analytics feature of Kong Enterprise. Find out more about Vitals [here](https://docs.konghq.com/enterprise/latest/vitals/overview/). |
> > |`enterprise.portal.enabled` | Set to `true` to enable Kong Enterprise's Developer Portal. Find out more about Vitals [here](https://docs.konghq.com/enterprise/latest/developer-portal/). |
> > |`enterprise.rbac.enabled` | Set to `true` to enable RBAC (role-based access control) for Kong Enterprise's Manager. |
> > ```

In the master values file, `enterprise.license_secret` already matches what is being configured in this guide, and `enterprise.vitals.enabled` is enabled by default. I think to keep things simple we don't have to list properties that are already correctly set.

`enterprise.portal.enabled` is a good point, let me add that one. I already have `enterprise.rbac.enabled`.

> 1. I would also order the parameters on the table based on the order of how they appear on the values.yaml file.

That's a good point. The only issue is that the `enterprise.*` values would appear at the very end of the table since they are quite at the end of the master values file - but these are the most relevant here. So I'd want to keep them at the top, but the rest of the table should follow the structure of the file.

**Comment by marcoam** [comment]
Thanks for working on this @cwurm - I like it :)

**Review by marcoam — APPROVED** [review]

File: app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 11]
```diff
@@ -147,26 +147,38 @@ In the following steps, replace `<your-password>` with a secure password.
...
     ```
-    $ oc create secret generic kong-session-config -n kong --from-file=admin_gui_session_conf=<manager-config-filename> --from-file=portal_session_conf=<portal-config-filename>
+    $ oc create secret generic kong-session-config -n kong --from-file=admin_gui_session_conf --from-file=portal_session_conf
     ```
 {% endnavtab %}
 {% endnavtabs %}
 ## Step 7. Prepare Kong's configuration file
 
-1. Create a `values.yaml` file based on the sample in the [Kong charts repository](https://github.com/Kong/charts/blob/main/charts/kong/values.yaml). This file sets all the necessary parameters for your Kong environment.
+1. Create a `values.yaml` file for Helm based on the template in the [Kong charts repository](https://github.com/Kong/charts/blob/main/charts/kong/values.yaml). This file contains all the necessary parameters for your Kong deployment. As an example, [this values file](https://github.com/Kong/charts/blob/main/charts/kong/example-values/full-k4k8s-with-kong-enterprise.yaml) enables most Kong Enterprise features.
```
Comment (lena-larionova) [inline app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md:156]:
```suggestion
1. Create a `values.yaml` file for Helm based on the template in the [Kong charts repository](https://github.com/Kong/charts/blob/main/charts/kong/values.yaml). This file contains all the possible parameters for your Kong deployment. 

    You can also base your configuration on a sample Kong Enterprise `values.yaml` 
    file. For example, [this values file](https://github.com/Kong/charts/blob/main/charts/kong/example-values/full-k4k8s-with-kong-enterprise.yaml) 
    enables most Kong Enterprise features.
```
I'd suggest a better lead-in for the Kong Enterprise sample file, to make it clear why they might want to use it. 

(I hope I got the indents right, it's difficult to judge in these suggestions.)

File: app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 12]
```diff
@@ -147,26 +147,38 @@ In the following steps, replace `<your-password>` with a secure password.
...
 
 2. Minimally, for setting up Kong Enterprise on Kubernetes, you will need to set the following parameters:
 
     |Parameter      | Value                         |
     |---------------|-------------------------------|
     |`enterprise.enabled` | `true` |
+    |`enterprise.portal.enabled` | (Optional) Set to `true` to enable the Kong Developer Portal. Find out more about it [here](https://docs.konghq.com/enterprise/latest/developer-portal/). |
```
Comment (lena-larionova) [inline app/enterprise/1.5.x/kong-for-kubernetes/install-on-kubernetes.md:163]:
```suggestion
    |`enterprise.portal.enabled` | (Optional) Set to `true` to enable the [Kong Developer Portal](/enterprise/{{page.kong_version}}/developer-portal/). |
```

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 13]
```diff
@@ -158,26 +158,38 @@ In the following steps, replace `<your-password>` with a secure password.
...
 
-1. Create a `values.yaml` file based on the sample in the [Kong charts repository](https://github.com/Kong/charts/blob/main/charts/kong/values.yaml). This file sets all the necessary parameters for your Kong environment.
+1. Create a `values.yaml` file for Helm based on the template in the [Kong charts repository](https://github.com/Kong/charts/blob/main/charts/kong/values.yaml). This file contains all the necessary parameters for your Kong deployment. As an example, [this values file](https://github.com/Kong/charts/blob/main/charts/kong/example-values/full-k4k8s-with-kong-enterprise.yaml) enables most Kong Enterprise features.
 
 2. Minimally, for setting up Kong Enterprise on Kubernetes, you will need to set the following parameters:
 
     |Parameter      | Value                         |
     |---------------|-------------------------------|
     |`enterprise.enabled` | `true` |
+    |`enterprise.portal.enabled` | (Optional) Set to `true` to enable the Kong Developer Portal. Find out more about it [here](https://docs.konghq.com/enterprise/latest/developer-portal/). |
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:174]:
```suggestion
    |`enterprise.portal.enabled` | (Optional) Set to `true` to enable the [Kong Developer Portal](/enterprise/{{page.kong_version}}/developer-portal/). |
```

**Review by lena-larionova — APPROVED** [review]
A couple of minor edits following Marco's suggestions, otherwise looks good!

**Pull request merged**